### PR TITLE
Add answers-core to dependencies

### PR DESCRIPTION
**Pull request by cea2aj**
Add @yext/answers-core to the dev dependencies

Also update the interface because provide core changed from async to sync.

Set browser: true for the resolve plugin so that rollup uses the browser ponyfill for cross-fetch rather than the node ponyfill.

Use the commonjs build to avoid the 'this is undefined' warning which appears because the core library outputs an awaiter helper which references `this` for the async answers core functions (universalSearch, verticalSearch, etc). The helpers are created because the core esm bundle targets an older version of JS than es2017.

Tests fail because I still need to update the core adapter to use the latest core api. That will be in a future PR.

J=SLAP-839
TEST=manual

Load a site with the SDK and confirm that Answers initializes

### CODE REVIEW

**Comment by oshi97** [comment]
oh in the commit message it says dev dependencies but it should just say dependencies

File: conf/gulp-tasks/bundle/bundle.js [hunk 1]
```diff
@@ -88,7 +94,13 @@ exports.legacyBundle = function (callback, outputConfig, bundleName, locale, lib
           '@babel/plugin-transform-object-assign'
         ]
       })
-    ]
+    ],
+    onwarn: function(warning) {
```
Comment (oshi97) [inline conf/gulp-tasks/bundle/bundle.js:98]: I think I would rather use the commonjs bundle than have this suppression, if the bundle size is basically the same

File: conf/gulp-tasks/bundle/bundle.js [hunk 2]
```diff
@@ -29,16 +29,22 @@ exports.modernBundle = function (callback, outputConfig, bundleName, locale, lib
...
       }),
       babel({
         babelrc: false,
-        exclude: [/node_modules/, /answers-core/],
+        exclude: [/node_modules/],
         presets: ['@babel/env']
       })
-    ]
+    ],
+    onwarn: function(warning) {
```
Comment (oshi97) [inline conf/gulp-tasks/bundle/bundle.js]: I think I'd rather use the commonjs bundle than have this onwarn, assuming the bundle sizes are basically the same. not sure what other people think

Comment (cea2aj) [reply conf/gulp-tasks/bundle/bundle.js]: ya, they're nearly the same. I'll update.

Leaving it as it is now would suppress the warning for other dependencies, and we probably don't want that

**Review by oshi97 — APPROVED** [review]

**Pull request merged**